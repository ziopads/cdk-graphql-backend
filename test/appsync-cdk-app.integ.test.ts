/**
 * Tests AppSyncCdkApp
 * 
 * @group e2e
 */
import Amplify, { API } from 'aws-amplify';

Amplify.configure({
  aws_appsync_region: process.env.AWS_DEFAULT_REGION, // Stack region
  aws_appsync_graphqlEndpoint: process.env.API_URL, // AWS AppSync endpoint
  aws_appsync_authenticationType: "API_KEY", //Primary AWS AppSync authentication type
  aws_appsync_apiKey: process.env.API_KEY // AppSync API Key
});

describe('List Empty Notes', () => {
  it('List Notes, should be empty', async () => {
    const query = `
      query listNotes {
        listNotes {
          id name completed
        }
      }
    `
    const response = await API.graphql({ query });
    expect(response).toMatchSnapshot();
  });
});

describe('Create Note', () => {
  it('Create Note', async () => {
    const mutation = `
      mutation createNote {
        createNote(note: {
          id: "001"
          name: "My note"
          completed: false
        }) {
          id
          name
          completed
        }
      }
    `
    const response = await API.graphql({
      query: mutation
    });
    expect(response).toMatchSnapshot();
  });
});

describe('Get Note By Id', () => {
  it('Get Note', async () => {
    const query = `
      query getNoteById {
        getNoteById(noteId: "001") {
          id
          name
          completed
        }
      }
    `
    const response = await API.graphql({ query });
    expect(response).toMatchSnapshot();
  });
});

describe('List Notes', () => {
  it('List Notes, should not be empty', async () => {
    const query = `
      query listNotes {
        listNotes {
          id name completed
        }
      }
    `
    const response = await API.graphql({ query });
    expect(response).toMatchSnapshot();
  });
});

describe('Update Note', () => {
  it('Update Note', async () => {
    const mutation = `
      mutation updateNote {
        updateNote(note: {
          id: "001"
          completed: true
        }) {
          id
          completed
        }
      }
    `
    const response = await API.graphql({
      query: mutation
    });
    expect(response).toMatchSnapshot();
  });
});

describe('Delete Note', () => {
  it('Delete Note', async () => {
    const mutation = `
      mutation deleteNote {
        deleteNote(noteId: "001")
      }
    `
    const response = await API.graphql({
      query: mutation
    });
    expect(response).toMatchSnapshot();
  });
});