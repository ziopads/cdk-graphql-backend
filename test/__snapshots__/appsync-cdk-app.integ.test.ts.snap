// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Note Create Note 1`] = `
Object {
  "data": Object {
    "createNote": Object {
      "completed": false,
      "id": "001",
      "name": "My note",
    },
  },
}
`;

exports[`Delete Note Delete Note 1`] = `
Object {
  "data": Object {
    "deleteNote": "001",
  },
}
`;

exports[`Get Note By Id Get Note 1`] = `
Object {
  "data": Object {
    "getNoteById": Object {
      "completed": false,
      "id": "001",
      "name": "My note",
    },
  },
}
`;

exports[`List Empty Notes List Notes, should be empty 1`] = `
Object {
  "data": Object {
    "listNotes": Array [],
  },
}
`;

exports[`List Notes List Notes, should not be empty 1`] = `
Object {
  "data": Object {
    "listNotes": Array [
      Object {
        "completed": false,
        "id": "001",
        "name": "My note",
      },
    ],
  },
}
`;

exports[`Update Note Update Note 1`] = `
Object {
  "data": Object {
    "updateNote": Object {
      "completed": true,
      "id": "001",
    },
  },
}
`;
